hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -63,34 +71,69 @@ public UniqueTokenStore(final EntityIdSource ids,
 	) {
 		super(ids, validator, properties, tokens, tokenRelsLedger);
 		this.uniqueTokenSupplier = uniqueTokenSupplier;
+		this.provisionalUniqueTokens = new ArrayList<>(); // TODO MAX_SIZE
+		this.lastMintedSerialNumbers = new ArrayList<>();
```

Comment:
Do we need those to be class properties if we are accessing them only in the `mint` method? they can be variables inside the mint method scope

Reply:
I'm accessing both in some methods. The ```provisionalUniqueTokens``` variable can be passed around as an argument. However, for ```lastMintedSerialNumbers``` , I can't think of other solution for now